src/Symfony/Component/FeatureFlag/Provider/InMemoryProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag\Provider;
+
+use Symfony\Component\FeatureFlag\ArgumentResolver\ArgumentResolver;
+
+/**
+ * @experimental
+ */
+final class InMemoryProvider implements ProviderInterface
+{
+    /**
+     * @param array<string, (\Closure(): mixed)> $features
+     */
+    public function __construct(
+        private readonly array $features,
+    ) {
+    }
+
+    public function has(string $featureName): bool
+    {
+        return \array_key_exists($featureName, $this->features);
+    }
+
+    public function get(string $featureName): \Closure
+    {
+        return $this->features[$featureName] ?? fn() => false;
```

Comment:
how do we track unknown features ? We don't have the fallback we previously had. Nor any log (which shouldn't break the app if the flag is unkown).

I also doubt that using this you can easily identify from the traceable provider that it was an unknown feature (to log warning in dev mode for example)

Reply:
The traceable provider uses the `getNames` method to determine if a feature exists.

In `dev` mode, we can see if a checked feature exists in the profiler:
![image](https://github.com/user-attachments/assets/7b21b2cf-aa2e-4cab-81d0-fb552fc72111)

Reply:
ok. What about returning null instead to make it more extendable in teh future ? Traceability, defaults, etc

Reply:
OK, but if `get` can return `null`, is the `has` method still useful?

Reply:
Yes because depending on future provider it might make things faster / more memory efficient to check if it has the feature before doing the compute.

Reply:
The feature resolution is not done when using `get` since it returns a Closure if it registered.

So we could reduce the combo `has(string $featureName): bool` + `get(string $featureName): \Closure` to `get(string $featureName): ?\Closure`. It could make some providers easier to implement.

Only the `FeatureChecker` will callback to `fn() => false` if no feature is found.

Reply:
👍 and it would allow a future default on feature checker 👍

Reply:
LGTM, it's done!